src/Cli/dotnet/ParseResultExtensions.cs

```diff
@@ -34,18 +34,31 @@ public static void ShowHelp(this ParseResult parseResult)
             Parser.Instance.Parse(tokenList).Invoke();
         }
 
+        private static string StripPlaceholdersFromString(string input)
+        {
+            // Find and remove all format placeholder strings (and filled runtime strings!) of the form ['"].*['"]
+            // from an input string.
+            // This lets us do fair comparisons against strings that are translated, since the placeholders
+            // may be in various locations, and may have different quote characters in different languages.
+            return System.Text.RegularExpressions.Regex.Replace(input, @"['""](.*?)['""]", string.Empty);
```

Comment:
I'm bad at regex, but doesn't this still assume that the quoting character is either a single or a double quote?

Would it be better to parse the format string into its components, and check that the error message contains the non-substituted components of the format string in the right order?

IE the string `Unrecognized command or argument '{0}'` has these components:

- `Unrecognized command or argument '` - literal
- '{0}' - substitution
- `'` - literal

So we would check that the error string started with the first component, and that the third component was found somewhere after the end of the first one.

That way I don't think we'd be dependent on localization or on the substituted arguments having quotes around them at all.

Reply:
Yeah, that's a solid plan. I'll work on that today.